coderd/templates_test.go

```diff
@@ -1326,3 +1327,57 @@ func TestTemplateMetrics(t *testing.T) {
 		dbtime.Now(), res.Workspaces[0].LastUsedAt, time.Minute,
 	)
 }
+
+func TestNotifyDeletedTemplate(t *testing.T) {
+	t.Parallel()
+
+	t.Run("OnlyNotifyOwnersAndTemplateAdmins", func(t *testing.T) {
+		t.Parallel()
+
+		var (
+			notifyEnq = &testutil.FakeNotificationsEnqueuer{}
+			client    = coderdtest.New(t, &coderdtest.Options{
+				IncludeProvisionerDaemon: true,
+				NotificationsEnqueuer:    notifyEnq,
+			})
+			firstUser = coderdtest.CreateFirstUser(t, client)
+			version   = coderdtest.CreateTemplateVersion(t, client, firstUser.OrganizationID, nil)
+			_         = coderdtest.AwaitTemplateVersionJobCompleted(t, client, version.ID)
+			template  = coderdtest.CreateTemplate(t, client, firstUser.OrganizationID, version.ID)
+			ctx       = testutil.Context(t, testutil.WaitLong)
+		)
+
+		_, anotherOwner := coderdtest.CreateAnotherUser(t, client, firstUser.OrganizationID, rbac.RoleOwner())
+		_, tmplAdmin := coderdtest.CreateAnotherUser(t, client, firstUser.OrganizationID, rbac.RoleTemplateAdmin())
+		coderdtest.CreateAnotherUser(t, client, firstUser.OrganizationID, rbac.RoleMember())
+
+		// When: template is deleted
+		err := client.DeleteTemplate(ctx, template.ID)
+		require.NoError(t, err)
+
+		// Then: the template owners and template admins should receive the notification
+		shouldBeNotified := []uuid.UUID{anotherOwner.ID, tmplAdmin.ID}
+		var deleteTemplateNotifications []*testutil.Notification
+		for _, n := range notifyEnq.Sent {
+			if n.TemplateID == notifications.TemplateTemplateDeleted {
+				deleteTemplateNotifications = append(deleteTemplateNotifications, n)
+			}
+		}
+		require.Len(t, deleteTemplateNotifications, len(shouldBeNotified))
+
+		for _, userID := range shouldBeNotified {
```

Comment:
I think you should invert this; you should find all the users which were notified and check against the expected list (you may need to sort the slice first).

This way, you don't miss any cases. I know you have the length check above, but this will be more explicit.